bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,377 @@
+            ]
+        else:
+            this_all_keys_dict[video_key] = [
+                False,  # NZSL Absent
+                True,  # S3 PRESENT
+                "",  # gloss_idgloss,
+                "",  # gloss_created_at,
+                "",  # gloss_id
+                "",  # video_id,
+                "",  # gloss_public,
+                "",  # video_public,
+            ]
+
+    # Find NZSL keys that are absent from S3 (present handled above)
+    for video_key, [
+        gloss_idgloss,
+        gloss_created_at,
+        gloss_id,
+        video_id,
+        gloss_public,
+        video_public,
+    ] in this_nzsl_raw_keys_dict.items():
+        if video_key not in this_s3_bucket_raw_keys_list:
+            this_all_keys_dict[video_key] = [
+                True,  # NZSL PRESENT
+                False,  # S3 Absent
+                gloss_idgloss,
+                gloss_created_at,
+                gloss_id,
+                video_id,
+                gloss_public,
+                video_public,
+            ]
+
+    return this_all_keys_dict
+
+
+# Cases
+# In S3     In NZSL     Action
+#   Is        Not         Delete S3 Object
+#   Is        Is          Update ACL
+#   Not       Is          Review
+def get_action(key_in_nzsl, key_in_s3):
+    if key_in_s3:
+        if key_in_nzsl:
+            return "Update ACL"
+        else:
+            return "Delete S3 Object"
+    return "Review"
+
+
+# Get S3 object's ACL
+def get_s3_canned_acl(video_key):
+    result = aws_cli(
+        [
+            "s3api",
+            "get-object-acl",
+            "--output",
+            "text",
+            "--query",
+            "Grants[*].Permission",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ]
+    )
+    acls_grants = result.stdout.strip().split("\t")
+
+    canned_acl = "unknown"
+    if len(acls_grants) > 1:
+        if acls_grants[0] == "FULL_CONTROL" and acls_grants[1] == "READ":
+            canned_acl = "public-read"
+    elif acls_grants[0] == "FULL_CONTROL":
+        canned_acl = "private"
+
+    return canned_acl
+
+
+# Get S3 object's LastModified date/time
+def get_s3_lastmodified(video_key):
+    result = aws_cli(
+        [
+            "s3api",
+            "head-object",
+            "--output",
+            "text",
+            "--query",
+            "LastModified",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ]
+    )
+    return result.stdout.strip()
+
+
+def build_csv_header():
+    return CSV_DELIMITER.join(
+        [
+            "S3 Video key",
+            "Sbank Gloss",
+            "Sbank Gloss created at",
+            "S3 LastModified",
+            "S3 Expected Canned ACL",
+            "S3 Actual Canned ACL",
+            "Sbank Gloss ID",
+            "Sbank Video ID",
+            "Sbank Gloss public",
+            "Sbank Video public",
+            "Action",
```

Comment:
issue: you've changed the order of these so they no longer match [what we agreed previously](https://github.com/ODNZSL/NZSL-signbank/pull/180#discussion_r1765668082)

The only potential change in order I think we should consider is in having "Action" as the first column since every item will have an action and that's arguably the most important thing to know

Reply:
```
issue: can we reorder these to:

S3 key
S3 Created At
Expected Canned ACL
Actual Canned ACL
Gloss ID
Video ID
Gloss Public
Video Public

```

Reply:
Apart from the additional fields `Gloss` and `LastModified` these do appear to be in that order?
I'll check what's being output to the spreadsheet.

Reply:
I'll move 'Action' to the first column.

Reply:
Done!